iXBRLViewerPlugin/__init__.py

```diff
@@ -143,8 +164,42 @@ def setLaunchIXBRLViewer(self, *args):
 
 def guiRun(cntlr, modelXbrl, attach, *args, **kwargs):
     """ run iXBRL Viewer using GUI interactions for a single instance or testcases """
-    if cntlr.hasGui and cntlr.launchIXBRLViewer.get() and modelXbrl.modelDocument.type in (Type.INLINEXBRL, Type.INLINEXBRLDOCUMENTSET):
-        launchLocalViewer(cntlr, modelXbrl)
+    class iXBRLViewerLocalViewer(LocalViewer):
+        # plugin-specific local file handler
+        def getLocalFile(self, file, relpath, request):
+            _report, _sep, _file = file.partition("/")
+            if file == 'ixbrlviewer.js':
+                return static_file('ixbrlviewer.js', os.path.abspath(os.path.join(os.path.dirname(__file__), "viewer", "dist")))
+            elif _report.isnumeric():  # in reportsFolder folder
+                # check if file is in the current or parent directory (may bve
+                _fileDir = self.reportsFolders[int(_report)]
+                _fileExists = False
+                if os.path.exists(os.path.join(_fileDir, _file)):
+                    _fileExists = True
+                elif "/" in _file and os.path.exists(os.path.join(_fileDir, os.path.filepart(_file))):
+                    # xhtml in a subdirectory for output files may refer to an image file in parent directory
+                    _fileExists = True
+                    _file = os.path.filepart(_file)
+                if not _fileExists:
+                    self.cntlr.addToLog("http://localhost:{}/{}".format(self.port, file), messageCode="localViewer:fileNotFound", level=logging.DEBUG)
+                return static_file(_file, root=_fileDir, headers=self.noCacheHeaders)  # extra_headers modification to py-bottle
+            return static_file(file, root="/")  # probably can't get here unless path is wrong
+    try:
+        import webbrowser
+        global tempViewer
+        tempViewer = tempfile.TemporaryDirectory()
```

Comment:
It looks like this never gets cleaned up. Perhaps instantiate this with a `with` statement?